node/src/components/network/behavior.rs

```diff
@@ -122,12 +124,54 @@ impl Behavior {
     }
 
     /// Initiates gossiping the given message.
-    pub(super) fn gossip(&mut self, message: GossipMessage) {
-        if let Err(error) = self.gossip_behavior.publish(&*TOPIC, message) {
+    pub(super) fn gossip(&mut self, message: GossipObject, max_gossip_message_size: u32) {
+        let data = match message.serialize(max_gossip_message_size as usize) {
+            Ok(encoded) => encoded,
+            Err(err) => {
+                warn!(?err, "{}: failed to serialize gossip message", self.our_id);
+                return;
+            }
```

Comment:
I don't think it's a good idea to fail to gossip silently. Consensus, for example, requires for gossiping to work, if we fail to deliver consensus messages silently then the network will stall.